src/lib/api/user/server/userApiClient.server.ts

```diff
@@ -28,3 +30,72 @@ export const fetchUserList = async (id: string): Promise<IUser[]> => {
     throw new Error('API response is invalid'); // TODO: display error page
   }
 };
+
+/**
+ * Interface representing the API response for the function to get a user by userId.
+ */
+export interface IGetUserByIdResponse {
+  /**
+   * The unique identifier of the user to get.
+   */
+  id: IUser['id'];
+
+  /**
+   * The name of the user.
+   */
+  name: IUser['name'];
+}
+
+/**
+ * The API response object type is  currently different from how it's defined in the web API document.
+ * We named current response object as {@link ITemporaryGetUserByIdResponse.}
+ * Once the API is fixed, we need to update the response object type of the function to {@link IGetUserByIdResponse}
+ * This issue is mentioned in the issue {@link https://github.com/nishiki-tech/nishiki-frontend/issues/255}
+ */
+export interface ITemporaryGetUserByIdResponse {
+  status: string;
+  statusCode: number;
+  body: {
+    userId: IUser['id'];
+    username: IUser['name'];
+  };
+}
+
+/**
+ * Fetch a user by userId.
+ * @param userId - The unique identifier of the user.
+ * @returns {Promise<Result<IGetCurrentUserResponse | string>>} - Id of current logged in user.
```

Comment:
The type should not be specified in TSDoc.

The description for this `@return` may be wrong. It should not only be "ID"